### ISSUE TITLE
Only run package tests if Python or other relevant files are modified

### ISSUE THREAD

**Issue by MaxGhenis** [PSLmodels/OG-USA#737]
e.g. currently docs-only changes trigger full package tests

**Comment by jdebacker** [comment]
This would be great (not just here, but in many repos).  I don't know how to implement.  Any suggestions?

**Comment by MaxGhenis** [comment]
Adding this to the GH Action should work:
```
on:
  push:
    paths:
      - '*.py'
```

**Comment by rickecon** [comment]
@MaxGhenis @jdebacker . Seems like we probably want tests to run upon changes to the following files and file types as well, in addition to `*.py`:
```
environment.yml
requirements.txt
*.pkl
*.csv
*.json
```
And I think we have a test that runs on the documentation when it is changed to make sure the Jupyter Book builds.

**Comment by rickecon** [comment]
This issue is resolved in PR #738.
